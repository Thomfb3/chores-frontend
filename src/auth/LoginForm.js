import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import AppAlert from "../common/AppAlert"
import TextField from '@mui/material/TextField';
import Paper from '@mui/material/Paper';
import Button from '@mui/material/Button';
import LoadingButton from '@mui/lab/LoadingButton';
import { Link } from "react-router-dom";

function LoginForm({ login }) {
    const history = useHistory();

    const [formData, setFormData] = useState({
        username: "",
        password: ""
    });

    const [loading, setLoading] = useState(false);

    const [formErrors, setFormErrors] = useState([]);
    console.log(formErrors);
    console.debug(
        "LoginForm",
        "login=", typeof login,
        "formData=", formData,
        "formErrors", formErrors,
    );

    async function handleSubmit(evt) {
        evt.preventDefault();
        setLoading(true);
        let result = await login(formData);
        if (result.success) {
            history.push("/");
        } else {
            setFormErrors(result.errors);
            setLoading(false);
        };
    };

    function handleChange(evt) {
        const { name, value } = evt.target;
        setFormData(data => ({ ...data, [name]: value }));
    };

    function reset() {
       setFormErrors([]);
    };

    return (
        <div className="Form">
            <Paper
                elevation={3}
                sx={{
                        width: '50%', paddingTop: '30px', margin: 'auto', marginTop: '50px', paddingBottom: '10px',
                        '@media (max-width:900px)': {
                            width: '90%'
                        }
                    }}>
                <form className="Form" onSubmit={handleSubmit}>
                    <div className="Form__box-title">Login</div>
                    <div className="Form-group">
                        <TextField
                            id="username"
                            label="Username"
                            className="Form-input"
                            variant="outlined"
                            placeholder="  "
                            name="username"
                            type="text"
                            value={formData.username}
                            onChange={handleChange}
                            sx={{ m: 1, width: '95%' }}
                            required
                        />
                    </div>
                    <div className="Form-group">
                        <TextField
                            id="password"
                            label="Password"
                            className="Form-input"
                            placeholder="  "
                            name="password"
                            type="password"
                            value={formData.password}
                            onChange={handleChange}
                            sx={{ m: 1, width: '95%' }}
                            required
                        />
                    </div>

                    {formErrors.length
                        ? <AppAlert severity="error" messages={formErrors} reset={reset} resetNeeded={true} />
                        : null}

                    <div className="Form-group" style={{ textAlign: 'right' }}>
                        <LoadingButton
                            sx={{ m: 2, backgroundColor: '#1193ff', borderRadius: '5px' }}
                            variant="contained"
                            type="submit"
                            loading={loading}
                            onSubmit={handleSubmit}
                        >
                            Login
                        </LoadingButton>
                    </div>
                </form>
                <small className='Form__footer'>* Required Fields</small>
            </Paper>
            <div className='Form__after'>
                <p>Don't have an account?</p>
                <Link to="/signup" style={{textDecoration:"none"}}>
                    <Button
                        sx={{ m: 2, backgroundColor: '#F98200', borderRadius: '5px', textDecoration: "none",
                            '&:hover': {
                                backgroundColor: '#f95c00'
                            },}}
                        variant="contained"
                    >
                        Signup
                    </Button>
                </Link>
            </div>
        </div>
    );
};

export default LoginForm;