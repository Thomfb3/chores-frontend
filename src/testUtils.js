import React from "react";
import UserContext from "./auth/UserContext";
import ChoreCommentContext from "./comments/ChoreCommentContext";

const demoUser = {
  _id: "userid1",
  username: "testuser1",
  firstName: "testfirst1",
  lastName: "testlast1",
  profileImage: "defaultProfile.jpg",
  allTimePoints: 1000,
  currentPoints: 200,
  role: "admin"
};

const testTeam = {
  _id: "testTeam",
  name: "testTeam"
};

const testTeamUsers = [
  {
    _id: "userid1",
    username: "testuser1",
    firstName: "testfirst1",
    lastName: "testlast1",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "admin"
  },
  {
    _id: "userid2",
    username: "testuser2",
    firstName: "testfirst2",
    lastName: "testlast2",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid3",
    username: "testuser3",
    firstName: "testfirst3",
    lastName: "testlast3",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  },
  {
    _id: "userid4",
    username: "testuser4",
    firstName: "testfirst4",
    lastName: "testlast4",
    profileImage: "defaultProfile.jpg",
    allTimePoints: 1000,
    currentPoints: 200,
    role: "user"
  }
];

let choreComments = [];

const UserProvider =
  ({ children,
    currentUser = demoUser,
    currentTeam = testTeam,
    currentTeamUsers = testTeamUsers,
    isAdmin = false
  }) => (
      <UserContext.Provider value={{ currentUser, currentTeam, currentTeamUsers }}>
        {children}
      </UserContext.Provider>
    );


const CommentProvider =
  ({ children,
    currentChoreComments = choreComments
  }) => (
      <ChoreCommentContext.Provider value={{ currentChoreComments }}>
        {children}
      </ChoreCommentContext.Provider>
    );

export { UserProvider, CommentProvider };

