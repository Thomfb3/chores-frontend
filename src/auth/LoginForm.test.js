import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import LoginForm from "./LoginForm";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider  currentUser={null}>
                <LoginForm  />
            </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});
