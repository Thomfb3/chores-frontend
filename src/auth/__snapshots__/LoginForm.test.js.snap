// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 css-5xxjkm-MuiPaper-root"
    >
      <form
        class="Form"
      >
        <div
          class="Form__box-title"
        >
          Login
        </div>
        <div
          class="Form-group"
        >
          <div
            class="MuiFormControl-root MuiTextField-root Form-input css-nkaxf5-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="username"
              id="username-label"
            >
              Username
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="username"
                name="username"
                placeholder="  "
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Username *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="Form-group"
        >
          <div
            class="MuiFormControl-root MuiTextField-root Form-input css-nkaxf5-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="false"
              for="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden="true"
                class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 *
              </span>
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                id="password"
                name="password"
                placeholder="  "
                required=""
                type="password"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    Password *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="Form-group"
          style="text-align: right;"
        >
          <button
            class="MuiLoadingButton-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-58dpzn-MuiButtonBase-root-MuiButton-root-MuiLoadingButton-root"
            tabindex="0"
            type="submit"
          >
            Login
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </form>
      <small
        class="Form__footer"
      >
        * Required Fields
      </small>
    </div>
    <div
      class="Form__after"
    >
      <p>
        Don't have an account?
      </p>
      <a
        href="/signup"
        style="text-decoration: none;"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1v3cooe-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Signup
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
